app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -847,6 +847,12 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
                 browserFragment.setImageBlockingEnabled(blockingImages);
             }
             setMenuButtonSelected(R.id.menu_blockimg, blockingImages);
+        } else if (this.getResources().getString(R.string.pref_key_allow_mixed_content).equals(key)) {
+            boolean allowMixedContent = Settings.getInstance(this).getAllowMixedContent();
+            BrowserFragment browserFragment = getBrowserFragment();
+            if (browserFragment != null) {
+                browserFragment.setAllowMixedContent(allowMixedContent);
+            }
         }
         // For turbo mode, a automatic refresh is done when we disable block image.
```

Comment:
https://github.com/mozilla-tw/Rocket/pull/2964/files/0c8cb9627501b40ef5e35bbb14a39153df1b9352#r236575804

I guess you mean this line of comment?
I think yes, we should remove this comment or at least move/merge it to first if statement.

Reply:
I actually mean
```
        // TODO: Better if we can move this logic to some setting-like classes, and provider interface
        // for configuring blocking function of each tab.
```

Reply:
Your statement is true too.